src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java

```diff
@@ -0,0 +1,219 @@
+    return new SimpleDateFormat(HistoricalDataConstants.EBD_TIME_FORMAT).format(new Date(time));
+  }
+
+  /**
+   * Get the current long time pointer.
+   *
+   * @return current time as long
+   */
+  private static long getCurrentTime() {
+    return System.currentTimeMillis();
+  }
+
+  /**
+   * Get the historical log data for all tag groups within the next FIFO queue time span.
+   *
+   * @param startNewTimeTracker if new time tracker should be generated, not read from storage
+   * @return historical log data
+   * @throws IOException if unable to read or write files
+   */
+  public static synchronized ArrayList getFifoNextSpanDataAllGroups(boolean startNewTimeTracker)
+      throws IOException {
+    final boolean includeTagGroupA = true;
+    final boolean includeTagGroupB = true;
+    final boolean includeTagGroupC = true;
+    final boolean includeTagGroupD = true;
+    return getFifoNextSpanData(
+        startNewTimeTracker,
+        includeTagGroupA,
+        includeTagGroupB,
+        includeTagGroupC,
+        includeTagGroupD);
+  }
+
+  /**
+   * Get the historical log data for the specified tag groups within the next FIFO queue time span.
+   *
+   * @param startNewTimeTracker if new time tracker should be generated, not read from storage
+   * @param includeTagGroupA if tag group A data should be included
+   * @param includeTagGroupB if tag group B data should be included
+   * @param includeTagGroupC if tag group C data should be included
+   * @param includeTagGroupD if tag group D data should be included
+   * @return historical log data
+   * @throws IOException if unable to read or write files
+   */
+  public static synchronized ArrayList getFifoNextSpanData(
+      boolean startNewTimeTracker,
+      boolean includeTagGroupA,
+      boolean includeTagGroupB,
+      boolean includeTagGroupC,
+      boolean includeTagGroupD)
+      throws IOException {
+    // Get start time from file, or start new time pointer if startNewTimePtr is true.
+    final String timeMarkerFileName =
+        HistoricalDataConstants.QUEUE_FILE_FOLDER
+            + HistoricalDataConstants.QUEUE_TIME_FILE_NAME
+            + HistoricalDataConstants.QUEUE_FILE_EXTENSION;
+    String startTimeTracker = "";
+    if (startNewTimeTracker) {
+      startTimeTracker = "" + getCurrentTime();
+      writeDataFile(timeMarkerFileName, startTimeTracker);
+    } else {
+      startTimeTracker = readDataFile(HistoricalDataConstants.FILE_URL_PREFIX + timeMarkerFileName);
+    }
+    final long startTimeTrackerLong = Long.parseLong(startTimeTracker);
+
+    // Calculate end time with FIFO time span
+    String endTimeTracker = "";
+    if (startTimeTrackerLong + getQueueFifoTimeSpanMillis() > getCurrentTime()) {
+      endTimeTracker = "" + getCurrentTime();
+    } else {
+      endTimeTracker = "" + (startTimeTrackerLong + getQueueFifoTimeSpanMillis());
+    }
+    final long endTimeTrackerLong = Long.parseLong(endTimeTracker);
```

Comment:
i think this method should get refactored so that the native long is maintained in scope to avoid this unnecessary conversion.

Reply:
refactored